pkg/check/chunkrepair/chunkrepair.go

```diff
@@ -0,0 +1,281 @@
+package chunkrepair
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/random"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+// Options represents chunk repair check options
+type Options struct {
+	NumberOfChunksToRepair int
+	Seed                   int64
+}
+
+// Check ...
+func Check(c bee.Cluster, o Options, pusher *push.Pusher, pushMetrics bool) (err error) {
+	ctx := context.Background()
+	rnds := random.PseudoGenerators(o.Seed, o.NumberOfChunksToRepair)
+	fmt.Printf("Seed: %d\n", o.Seed)
+
+	pusher.Collector(repairedCounter)
+	pusher.Collector(repairedTimeGauge)
+	pusher.Collector(repairedTimeHistogram)
+
+	pusher.Format(expfmt.FmtText)
+
+	for i := 0; i < o.NumberOfChunksToRepair; i++ {
+		// Pick node A, B, C and a chunk which is closest to B
+		nodeA, nodeB, nodeC, chunk, err := getNodes(ctx, c, rnds[i])
+		if err != nil {
+			return err
+		}
+		addressA, err := nodeA.Overlay(ctx)
+		if err != nil {
+			return err
+		}
+
+		// upload the chunk in nodeA
+		err = nodeA.UploadChunk(ctx, chunk)
+		if err != nil {
+			return err
+		}
+
+		for {
+			// check if the node is there in the local store of node B
+			// this does a get chunk instead of Has chunk, so the following
+			// call just checks if the chunk is accessible from nodeB
+			present, err := nodeB.HasChunk(ctx, chunk.Address())
+			if err != nil {
+				fmt.Println("sleeping...")
+
+				// give time for the chunk to reach its destination
+				time.Sleep(100 * time.Millisecond)
+				continue
+			}
+
+			if present {
+				break
+			}
+		}
+
+		// download the chunk from nodeC
+		data1, err := nodeC.DownloadChunk(ctx, chunk.Address(), "")
+		if err != nil {
+			return err
+		}
+		if !bytes.Equal(data1, chunk.Data()) {
+			return errors.New("chunk downloaded in NodeC does not have proper data")
+		}
+
+		// delete the chunk from all nodes. If the chunk from nodeA is not deleted,
+		// it is hard to simulate the chunk failure in small clusters. We would need a
+		// fairly large cluster then.
+		err = deleteChunkFromAllNodes(ctx, c, chunk)
+		if err != nil {
+			return err
+		}
+
+		// trigger downloading of the chunk from nodeC again (this time it should trigger chunk repair)
+		_, err = nodeC.DownloadChunk(ctx, chunk.Address(), addressA.String()[0:2])
+		if err == nil { // for status Accepted (209), a error is returned
```

Comment:
An error condition is when the error value is nil, but in that case, the nil error is returned.

Check for the expected error and return an appropriate error stating what is the problem, instead just nil.